qiskit/circuit/library/__init__.py

```diff
@@ -85,6 +117,9 @@
     This summary table deliberately does not generate toctree entries; these directives are "owned"
     by ``qiskit.circuit``.
 
+Directives are instructions to the quantum stack that are not reversible. They are all interpreted
```

Comment:
I'm wondering how how accurate this is, not all of these are something all backends understand or support.

Reply:
Good point. In c3c72d1a7, I rephrase it as:
> Directives are operations to the quantum stack that are meant to be intepreted by the backend.

Reply:
I think we need to rename and/or split this section. `Measure` and `Reset` aren't directives their operations (they're just not gates). The description you're proposing makes sense for measurement and reset  but not with the word directive. While `Barrier` is a directive but it's different because it's a compiler hint and has no meaning on the backend (similar examples would be save and set state instructions in aer).